// @studydefi/money-legos/dist/erc20/contracts.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var _a, _b, _c, _d, _e, _f, _g;
Object.defineProperty(exports, "__esModule", { value: true });
var ERC20_json_1 = __importDefault(require("./abi/ERC20.json"));
var networks_1 = __importDefault(require("../networks"));
var contracts = {
    abi: ERC20_json_1.default,
    bat: {
        address: (_a = {},
            _a[networks_1.default.mainnet] = "0x0d8775f648430679a709e98d2b0cb6250d2887ef",
            _a),
    },
    dai: {
        address: (_b = {},
            _b[networks_1.default.mainnet] = "0x6b175474e89094c44da98b954eedeac495271d0f",
            _b),
    },
    rep: {
        address: (_c = {},
            _c[networks_1.default.mainnet] = "0x1985365e9f78359a9B6AD760e32412f4a445E862",
            _c),
    },
    sai: {
        address: (_d = {},
            _d[networks_1.default.mainnet] = "0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359",
            _d),
    },
    usdc: {
        address: (_e = {},
            _e[networks_1.default.mainnet] = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
            _e),
    },
    wbtc: {
        address: (_f = {},
            _f[networks_1.default.mainnet] = "0x2260fac5e5542a773aa44fbcfedf7c193bc2c599",
            _f),
    },
    zrx: {
        address: (_g = {},
            _g[networks_1.default.mainnet] = "0xE41d2489571d322189246DaFA5ebDe1F4699F498",
            _g),
    },
};
exports.default = contracts;
