// @studydefi/money-legos/dist/compound/contracts.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
Object.defineProperty(exports, "__esModule", { value: true });
var Comptroller_json_1 = __importDefault(require("./abi/Comptroller.json"));
var CompoundPriceOracle_json_1 = __importDefault(require("./abi/CompoundPriceOracle.json"));
var CToken_json_1 = __importDefault(require("./abi/CToken.json"));
var CEther_json_1 = __importDefault(require("./abi/CEther.json"));
var networks_1 = __importDefault(require("../networks"));
var contracts = {
    cTokenAbi: CToken_json_1.default,
    comptroller: {
        address: (_a = {},
            _a[networks_1.default.mainnet] = "0x3d9819210a31b4961b30ef54be2aed79b9c9cd3b",
            _a),
        abi: Comptroller_json_1.default,
    },
    priceOracle: {
        address: (_b = {},
            _b[networks_1.default.mainnet] = "0x1d8aedc9e924730dd3f9641cdb4d1b92b848b4bd",
            _b),
        abi: CompoundPriceOracle_json_1.default,
    },
    cEther: {
        address: (_c = {},
            _c[networks_1.default.mainnet] = "0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5",
            _c),
        abi: CEther_json_1.default,
    },
    cDAI: {
        address: (_d = {},
            _d[networks_1.default.mainnet] = "0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643",
            _d),
        abi: CToken_json_1.default,
    },
    cSAI: {
        address: (_e = {},
            _e[networks_1.default.mainnet] = "0xf5dce57282a584d2746faf1593d3121fcac444dc",
            _e),
        abi: CToken_json_1.default,
    },
    cBAT: {
        address: (_f = {},
            _f[networks_1.default.mainnet] = "0x6c8c6b02e7b2be14d4fa6022dfd6d75921d90e4e",
            _f),
        abi: CToken_json_1.default,
    },
    cZRX: {
        address: (_g = {},
            _g[networks_1.default.mainnet] = "0xb3319f5d18bc0d84dd1b4825dcde5d5f7266d407",
            _g),
        abi: CToken_json_1.default,
    },
    cUSDC: {
        address: (_h = {},
            _h[networks_1.default.mainnet] = "0x39aa39c021dfbae8fac545936693ac917d5e7563",
            _h),
        abi: CToken_json_1.default,
    },
    cREP: {
        address: (_j = {},
            _j[networks_1.default.mainnet] = "0x158079ee67fce2f58472a96584a73c7ab9ac95c1",
            _j),
        abi: CToken_json_1.default,
    },
    cWBTC: {
        address: (_k = {},
            _k[networks_1.default.mainnet] = "0xc11b1268c1a384e55c48c2391d8d480264a3a7f4",
            _k),
        abi: CToken_json_1.default,
    },
};
exports.default = contracts;
