// @studydefi/money-legos/dist/maker/contracts.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var _a, _b, _c, _d, _e;
Object.defineProperty(exports, "__esModule", { value: true });
var ProxyRegistry_json_1 = __importDefault(require("./abi/ProxyRegistry.json"));
var DssCdpManager_json_1 = __importDefault(require("./abi/DssCdpManager.json"));
var DssProxyActions_json_1 = __importDefault(require("./abi/DssProxyActions.json"));
var networks_1 = __importDefault(require("../networks"));
var contracts = {
    proxyRegistry: {
        address: (_a = {},
            _a[networks_1.default.mainnet] = "0x4678f0a6958e4D2Bc4F1BAF7Bc52E8F3564f3fE4",
            _a),
        abi: ProxyRegistry_json_1.default,
    },
    dssCdpManager: {
        address: (_b = {},
            _b[networks_1.default.mainnet] = "0x5ef30b9986345249bc32d8928B7ee64DE9435E39",
            _b),
        abi: DssCdpManager_json_1.default,
    },
    dssProxyActions: {
        address: (_c = {},
            _c[networks_1.default.mainnet] = "0x82ecd135dce65fbc6dbdd0e4237e0af93ffd5038",
            _c),
        abi: DssProxyActions_json_1.default,
    },
    jug: {
        address: (_d = {},
            _d[networks_1.default.mainnet] = "0x19c0976f590D67707E62397C87829d896Dc0f1F1",
            _d),
    },
    daiJoin: {
        address: (_e = {},
            _e[networks_1.default.mainnet] = "0x9759A6Ac90977b93B58547b4A71c78317f391A28",
            _e),
    },
};
exports.default = contracts;
