// @studydefi/money-legos/dist/maker/ilks.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var _a, _b, _c, _d;
Object.defineProperty(exports, "__esModule", { value: true });
var erc20_1 = __importDefault(require("../erc20"));
var networks_1 = __importDefault(require("../networks"));
var ilks = {
    batA: {
        symbol: "BAT-A",
        token: { address: erc20_1.default.contracts.bat.address },
        join: {
            address: (_a = {},
                _a[networks_1.default.mainnet] = "0x3D0B1912B66114d4096F48A8CEe3A56C231772cA",
                _a),
        },
    },
    ethA: {
        symbol: "ETH-A",
        token: {
            address: (_b = {},
                _b[networks_1.default.mainnet] = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
                _b),
        },
        join: {
            address: (_c = {},
                _c[networks_1.default.mainnet] = "0x2F0b23f53734252Bda2277357e97e1517d6B042A",
                _c),
        },
    },
    usdcA: {
        symbol: "USDC-A",
        token: { address: erc20_1.default.contracts.usdc.address },
        join: {
            address: (_d = {},
                _d[networks_1.default.mainnet] = "0xA191e578a6736167326d05c119CE0c90849E84B7",
                _d),
        },
    },
};
exports.default = ilks;
